parser/block.go

```diff
@@ -44,13 +44,14 @@ func (b *Block) GetEncodableHash() []byte {
 }
 
 func (b *Block) GetDisplayPrevHash() []byte {
-	h := b.hdr.HashPrevBlock
+	rhash := make([]byte, len(b.hdr.HashPrevBlock))
+	copy(rhash, b.hdr.HashPrevBlock)
```

Comment:
We're passing in a pointer to a `Block` here, so we need to reverse (and return) a copy of the prev hash, not the actual prev hash! It would be weird if a method with "`Get`" in its name modified its argument! I found this while writing tests; I don't know if it affects production code.

Reply:
This looks good. See comment about endless reorg bug due to this code change